### ISSUE TITLE
JAWS reads soft hyphens in web content

### ISSUE THREAD

**Issue by Michael-Detmers** [FreedomScientific/standards-support#176]
### Summary

Pronounciation can be way off when JAWS is trying to process [soft hyphens](https://en.wikipedia.org/wiki/Soft_hyphen) it should ignore in the first place. 

Example:
Save the following HTML and try to read is via JAWS.
```HTML
<!DOCTYPE html>
<html lang="en">
  <head>
    <title>Hyphenation Test</title>
  </head>
  <body>
  <h1>Hooray for Hyphen&shy;ation</h1>
  <p>This text contains some Hyphen&shy;action. Hope&shy;fully it is not in&shy;com&shy;pre&shy;hen&shy;sible.</p>
  </body>
</html>
```

### Expected result

Soft hyphens should be ignored.

### Actual result

The words are audibly distorted by slicing them at the hyphen.

#### JAWS version and build number
J2019.1901.66

#### Browser and version:
For example Firefox 65.0.2.

**Comment by JAWS-test** [comment]
The problem also occurs with IE 11 and Chrome.
The difference becomes clearer when the character is positioned in the wrong places: <code>Hop&shy;efully it is not i&shy;ncomp&shy;rehe&shy;nsi&shy;ble</code>

**Comment by jducrot** [comment]
I used https://cdpn.io/TPG/debug/qBbRNPJ to test this issue and as of today this is still a valid issue. It's happening in Chrome 83, FF 77, IE 11.00. It's also happening in all 3 browsers using NVDA 2020.1.

**Comment by JAWS-test** [comment]
The problem persists

**Comment by TPGJRogers** [comment]
Bugs 123870 and 123871

**Comment by stevefaulkner** [comment]
tested with JAWS 2023.2307.37 AND CHROME Version 116.0.5845.141 (Official Build) (64-bit) WORKS AS EXPECTED

**Comment by TPGJRogers** [comment]
@stevefaulkner, I agree the pronunciation is far better than it was, so significant progress has been made as the bug was written. However, there are still remnants of this issue pertaining to braille. If I open a file as described by @Michael-Detmers in the initial report, there are soft hyphens present in the places listed below.
- Between the n and a in the string, “Hooray for Hyphenation”
- In the string, “This text contains some Hyphenaction. Hopefully it is not incomprehensible.”
  - Between the n and a in Hyphenaction
  - Between the e and f in Hopefully
  - Between the n and c, m and p, e and h, and n and s in incomprehensible

They no longer impact pronunciation of the words, but they do appear on the braille display. They are also presented as “empty places” when using arrow keys to move character-by-character through the word. If the soft hyphens are not presented visually, I contend their presence on the braille display, and in character-by-character review of the words, to still be a JAWS bug.

**Comment by stevefaulkner** [comment]
> @stevefaulkner, I agree the pronunciation is far better than it was, so significant progress has been made as the bug was written. However, there are still remnants of this issue pertaining to braille. If I open a file as described by @Michael-Detmers in the initial report, there are soft hyphens present in the places listed below.
> 
> * Between the n and a in the string, “Hooray for Hyphenation”
> * In the string, “This text contains some Hyphenaction. Hopefully it is not incomprehensible.”
>   
>   * Between the n and a in Hyphenaction
>   * Between the e and f in Hopefully
>   * Between the n and c, m and p, e and h, and n and s in incomprehensible
> 
> They no longer impact pronunciation of the words, but they do appear on the braille display. They are also presented as “empty places” when using arrow keys to move character-by-character through the word. If the soft hyphens are not presented visually, I contend their presence on the braille display, and in character-by-character review of the words, to still be a JAWS bug.

@BrettLewisVispero  braille output is still an issue, announced text is fixed

**Comment by JeremiahRogersVispero** [comment]
Bug 2101
